browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -51,9 +54,10 @@ WebUIController* NewWebUI<BasicUI>(WebUI* web_ui, const GURL& url) {
 WebUIFactoryFunction GetWebUIFactoryFunction(WebUI* web_ui,
                                              const GURL& url) {
   if (url.host_piece() == kPaymentsHost ||
+      url.host_piece() == kWelcomeHost ||
       url.host_piece() ==  chrome::kChromeUINewTabHost) {
     return &NewWebUI<BasicUI>;
-  } else if (url.spec() == kWelcomeRemoteURL) {
+  } else if (url.spec() == kBraveUIWelcomeURL) {
```

Comment:
I think this if clauses isn't used anymore because of above `url.host_piece() == kWelcomeHost`.
Does BraveWelcomeUIBrowserTest.StartupURLTest pass?

Reply:
I assume you didn't try to browser test yet.
You need to replace `kWelcomeRemoteURL`  with `kBraveUIWelcomeURL` in the test code.

To pass the test, logic in `BraveWelcomeUI::BraveWelcomeUI()` should be used.
How about make `BraveWelcomeUI` inherit `BasicUI`?

Reply:
I tested this PR on my local machine.
First, `BraveWelcomeUIBrowserTest` was failed in `PRE_StartupURLTest` because actual url isn't same as`kBraveUIWelcomeURL`.
The reason is reverse handler(`HandleURLReverseRewrite`) isn't working properly.
If reverse handler doesn't return true, virtual url is not updated with our ones.
Because of this, virtual url of tab at 0 index in test browser was different.
To fix this, `kBraveUIWelcomeURL` should be `chrome://welcome/` not `chrome://welcome`. (`url.spec()` is `chrome://welcome/`). Or checking like above newtab seems better instead of comparing whole url.

When run test again, test will be failed in `StartupURLTest` because `prefs::kHasSeenWelcomePage` and `prefs::kHasSeenWin10PromoPage` isn't set properly.
If these prefs are not updated, browser always create welcome ui tab at startup.
It was handled in ctor of `BraveWelcomeUI`.
So, I recommend that making BraveWelcomeUI as a subclass of `BasicUI` and use it.